Dockerfile

```diff
@@ -54,13 +53,19 @@ COPY Makefile /parca-agent
 COPY bpf /parca-agent/bpf
 RUN make -C bpf setup
 # hadolint ignore=DL3059
-RUN make bpf
+RUN if [ "${TARGETARCH}" = "amd64" ]; then \
+        make bpf CPPFLAGS='--target=x86_64-unknown-linux-gnu'; \
+    else \
+        make bpf CPPFLAGS="--target=${TARGETARCH}-unknown-linux-gnu"; \
+    fi
 
 COPY . /parca-agent
 RUN git submodule init && git submodule update
 RUN make build
 
-FROM docker.io/debian:bullseye-slim@sha256:06a93cbdd49a265795ef7b24fe374fee670148a7973190fb798e43b3cf7c5d0f AS all
+
+
+FROM --platform="${TARGETPLATFORM:-linux/amd64}" docker.io/debian:bullseye-slim@sha256:06a93cbdd49a265795ef7b24fe374fee670148a7973190fb798e43b3cf7c5d0f AS all
```

Comment:
Just a note about `--platform`, `docker-buildx` uses image for the `TARGETPLATFORM` in `FROM` by default, `podman-build` needs it to be explicit otherwise it uses the `BUILDPLATFORM`